src/test/rgw/bucket_notification/api.py

```diff
@@ -247,13 +247,23 @@ def delete_all_topics(conn, tenant, cluster):
     if tenant == '':
         topics_result = admin(['topic', 'list'], cluster)
         topics_json = json.loads(topics_result[0])
-        for topic in topics_json['topics']:
-            rm_result = admin(['topic', 'rm', '--topic', topic['name']], cluster)
-            print(rm_result)
+        if 'topics' not in topics_json:
+            topics_json = topics_json.get('result',{})
+        try:
+            for topic in topics_json['topics']:
+                rm_result = admin(['topic', 'rm', '--topic', topic['name']], cluster)
+                print(rm_result)
+        except KeyError as e:
```

Comment:
please do not catch the exception here.
if the results dont have `['topics']` and dont have `['result']['topics']` it is better to let the `KeyError` not be caught and fail the test

Reply:
Thank you for the feedback! I’ve removed the try-except block. If the keys `topics` or `result` are missing, the `KeyError` will now surface naturally, the test will fail appropriately.